### ISSUE TITLE
Missing variables for Form 1040

### ISSUE THREAD

**Issue by mmessick** [open-source-economics/Tax-Calculator#295]
For a number of lines in Form 1040, we are missing variables to deal with them. Usually these are lines that we do not have PUF data for, so we don't bother including them in calculations. However, once we start taking in individual information, we will want variables for these if user's have non-zero values listed on their forms. Right now I am just adding them to the bottom of the tax_ecodes_v2 spreadsheet, but can also list here which lines need corresponding variables. We might even need some for whole forms eventually (such as the W-2 and 1099).

**Comment by MattHJensen** [comment]
@zrisher, do you think this issue can be closed now?

**Comment by zrisher** [comment]
@MattHJensen I believe there is a substantial amount of tax information available on individual IRS forms that `tax-calc` doesn't take as input. This project currently simulates actual tax calculation with a subset of individual tax data because:
* we only have a subset of actual tax data for the general populace (`puf.csv`)
* mirroring the entirety of actual tax calculation is a lot of work, especially when you want to give users the ability to modify its logic via policy params (`current_law_policy.json`)

This approach makes sense for a tool with the primary purpose of studying how changes to tax policy would affect the nation as a whole. It's a "model." If `taxcalc` was designed to help you prepare your individual taxes, you'd want to build out all the individual parts, but you wouldn't care about making that logic easy to modify.

In my opinion, the current approach is the only viable option given the needs of most `tax-calc` users. Any model will work this way. It is completely possible to continue to break out individual parts of calculation here and there, but it must be made to work with the existing limited data we have for the nation as a whole.

**TLDR** There is work that could be done here, but it would only really benefit users with full tax return data. Not sure how you want to document that. You could generalize this issue's label, or you could simply make a point in the contributor docs that building out the model to deal with every bit of info from IRS forms is valuable (if that's actually desired).

@martinholmer @feenberg @GoFroggyRun @codykallen @Amy-Xu

**Comment by feenberg** [comment]
On Sat, 27 Aug 2016, zrisher wrote:

> 
> @MattHJensen I believe there is a substantial amount of tax information available on
> individual IRS forms that tax-calc doesn't take as input. This project currently
> simulates actual tax calculation with a subset of individual tax data because:

taxcalc.sas uses much more data, any internal government user of the 
python calculator would have that additional data.

It isn't so hard to add additional variables, but without data to test 
them  it might be foolish. It is possible one agency might take an 
interest in the calculator and allow testing on the real data.

Also, taxcalc.sas has been tested, and I could generate random data and 
compare sas and python results. That might be something to start with.

dan

**Comment by zrisher** [comment]
I said:
> you could simply make a point in the contributor docs that building out the model to deal with every bit of info from IRS forms is valuable (if that's actually desired).

We now have that model in `TaxForm`, so the question becomes how to best expose that additional data from a `TaxForm` to a taxcalc function, which currently only have access to the `VALID_READ_VARS` from their `Records`.

**Comment by MattHJensen** [comment]
Closing this issue until users demand the additional variables. At that point we can add them on a case by case basis.
